staging/src/kubevirt.io/api/core/v1/types.go

```diff
@@ -2163,6 +2176,17 @@ type RemoveVolumeOptions struct {
 	DryRun []string `json:"dryRun,omitempty"`
 }
 
+// AddInterfaceOptions is provided when dynamically hot plugging a network interface
+type AddInterfaceOptions struct {
+	// NetworkName references a NetworkAttachmentDefinition CRD object. Format:
+	// <networkName>, <namespace>/<networkName>. If namespace is not
+	// specified, VMI namespace is assumed.
+	NetworkName string `json:"networkName"`
```

Comment:
Be that as it may - and I think you're right - we should strive for ensuring cohesion across the API.

Independently of whether or not the names on the API are wrong / misleading, we should keep them consistent, and move all of them to a better name once we have them.

Fixing part of the API, imo, will only generate confusion to the users. I think this should happen, both for hotplug and for VMI spec interfaces / networks, in a follow-up PR.

Reply:
As you say, we have already discussed and addressed this in the design document: https://github.com/kubevirt/community/pull/201#discussion_r1042342131

You approved it afterwards: https://github.com/kubevirt/community/pull/201#pullrequestreview-1209677072

My point still holds: I think we should follow the principle of least astonishment, and feature consistent names throughout the API, instead of having different names (referring to the same thing) scattered throughout it.

Reply:
> There is a good chance I already commented about this in the design doc, my apologies if I repeat myself.
> 
> The existing VM/VMI spec for networking is, sub optimal, and creates a lot of confusion. I see no reason to repeat the confusion here. IMO, we should use `network` to refer to the VM/VMI spec network, which includes the `interface` and `network` there. And use `NetworkAttachementDefenition` to refer to the NAD. A NAD is _not_ a network, it is a connection definition to a network. The VMI interface and its network are actually the same thing. However, due to the naming mess, the interface is collision with the pod interface, and the network collides with the NAD.
> 
> IMO we should have here a clear name to what these things are referring. E.G. :
> 
> * VMNetworkName
> * NetworkAttachementDefinitionName

@EdDev hi，I'm curious about the specific situations causing these confusions. Could you provide some issues or other pages? Thank you in advance.